### ISSUE TITLE
On-premises kubernetes cluster

### ISSUE THREAD

**Issue by muhammed-sumer** [microsoft/mindaro#44]
Hi @greenie-msft and all team, 

When can we start using it for other clusters(without aks) and also when will it be released Bridge-To-Kubernetes for other clusters? 
<a href="https://code.visualstudio.com/docs/containers/bridge-to-kubernetes">Note Although this quickstart works with Azure Kubernetes Service (AKS), you can also try Bridge to Kubernetes with other Kubernetes clusters. Support for other clusters is in preview.</a>

I tried using it for preview for the on-premises kubernetes cluster, but I got an error like the following. What is the reason?

---

Debugging with Bridge to Kubernetes requires administrator permissions to:
 -   Update your machine's hosts file to match your Kubernetes cluster environment.
 
Once your cluster environment is replicated, all processes on your machine will be able to access it.

---
after saying continue...

[Window Title]
Visual Studio Code
The preLaunchTask 'bridge-to-kubernetes.compound' terminated with exit code 1.
Remember my choice in user settings [Debug Anyway] [Show Errors] [Abort]

---

Env. Details;
Windows 10 Pro - Visual Studio Code v1.49.3 - Bridge to Kubernetes v1.0.120200926

Thank you..

**Comment by pragyamehta** [comment]
Hi @muhammed-sumer , thanks for reaching out to us! 
Can you please provide the logs from "%temp%\Bridge To Kubernetes" folder?

**Comment by muhammed-sumer** [comment]
Hi @pragyamehta, I added the log files.  Thanks.

[bridge-mindarocli-2020-10-07-06-18-54-13848.txt](https://github.com/microsoft/mindaro/files/5338725/bridge-mindarocli-2020-10-07-06-18-54-13848.txt)
[bridge-mindarocli-2020-10-07-06-18-58-1644.txt](https://github.com/microsoft/mindaro/files/5338727/bridge-mindarocli-2020-10-07-06-18-58-1644.txt)
[bridge-library-2020-10-07-06-18-55-13848.txt](https://github.com/microsoft/mindaro/files/5338728/bridge-library-2020-10-07-06-18-55-13848.txt)
[bridge-library-2020-10-07-06-18-58-1644.txt](https://github.com/microsoft/mindaro/files/5338729/bridge-library-2020-10-07-06-18-58-1644.txt)

**Comment by riker09** [comment]
I've tried to debug the BikeSharingApp running inside a Docker for Desktop cluster. I can see that the redirectional pods are started, however, it fails with this error:

```
Failed to pull image "bridgetokubernetes.azurecr.io/lpkremoteagent:0.1.4": rpc error: code = Unknown desc = Error response from daemon: Get https://bridgetokubernetes.azurecr.io/v2/lpkremoteagent/manifests/0.1.4: unauthorized: authentication required, visit https://aka.ms/acr/authorization for more information.
```

The image `lpkremoteagent:0.1.4` seems to be hosted on a private Azure Container Registry, is that correct? Shouldn't that be public?

**Comment by jamyc** [comment]
@riker09 Running into the same issue, this wasn't the case a few days ago. Created a separate issue #45 since this doesn't seem to be related to on-prem clusters.

**Comment by greenie-msft** [comment]
@riker09, @jamyc - The "Failed to pull image" issue has been resolved. Please let us know if you encounter any other problems.

@muhammed-sumer  - Thank you for providing your logs. The team is actively looking into the issue and will provide an update as soon as we have more information. 

Thank you,
Nick

**Comment by riker09** [comment]
Thank you for your quick response on this. I will check it out tomorrow. May I ask what has been changed?

**Comment by pragyamehta** [comment]
@riker09  Our images reside in an azure container registry, and we had to update the ACR such that our customers could pull images from it successfully.

**Comment by pragyamehta** [comment]
@muhammed-sumer We have released the fix for your issue. Please try again and let us know if you are unblocked now!

**Comment by muhammed-sumer** [comment]
Hi @pragyamehta, That's great. It worked in local environment after the update. I will also try it on the internal kubernetes server. I will share my experiences with you again. Thank you..

**Comment by daniv-msft** [comment]
Closing the issue. Thanks for reporting it!
